engine.py

```diff
@@ -190,111 +190,59 @@ def host_info(self):
-            try:
-                from PyQt4 import QtCore, QtGui
-                import PyQt4
-                
-                # tell QT to interpret C strings as utf-8
-                utf8 = QtCore.QTextCodec.codecForName("utf-8")
-                QtCore.QTextCodec.setCodecForCStrings(utf8)                
-                
-                # a simple dialog proxy that pushes the window forward
-                class ProxyDialogPyQt(QtGui.QDialog):
-                    def show(self):
-                        QtGui.QDialog.show(self)
-                        self.activateWindow()
-                        self.raise_()
-                
-                    def exec_(self):
-                        self.activateWindow()
-                        self.raise_()
-                        # the trick of activating + raising does not seem to be enough for
-                        # modal dialogs. So force put them on top as well.                        
-                        self.setWindowFlags(QtCore.Qt.WindowStaysOnTopHint | self.windowFlags())
-                        return QtGui.QDialog.exec_(self)
-                
-                
-                # hot patch the library to make it work with pyside code
-                QtCore.Signal = QtCore.pyqtSignal
-                QtCore.Slot = QtCore.pyqtSlot
-                QtCore.Property = QtCore.pyqtProperty             
-                base["qt_core"] = QtCore
-                base["qt_gui"] = QtGui
-                base["dialog_base"] = ProxyDialogPyQt
-                self.log_debug("Successfully initialized PyQt '%s' located in %s." 
-                               % (QtCore.PYQT_VERSION_STR, PyQt4.__file__))
-                self._has_qt = True
-            except ImportError:
-                pass
-            except Exception, e:
-                self.log_warning("Error setting up PyQt. PyQt based UI support will not "
-                                 "be available: %s" % e)
-        
+        base = super(ShellEngine, self)._define_qt_base()
+
+        if not base["qt_gui"]:
+            self._has_qt = False
+
+            # proxy class used when QT does not exist on the system.
+            # this will raise an exception when any QT code tries to use it
+            class QTProxy(object):
+                def __getattr__(self, name):
+                    raise tank.TankError(
+                        "The Shotgun Toolkit App you are trying to execute requires a full QT "
+                        "environment in order to render its UI. A valid PySide2/PySide/PyQt "
+                        "installation could not be found in your python system path."
+                    )
+
+            base = {"qt_core": QTProxy(), "qt_gui": QTProxy(), "dialog_base": None}
+
+        else:
+            self._has_qt = True
+            QtCore = base["qt_core"]
+            QtGui = base["qt_gui"]
+
+            # tell QT4 to interpret C strings as utf-8
```

Comment:
Idk if this is overkill, but maybe make the comment a little more explicit about why this only affects Qt4?

```suggestion
            # Tell QT4 to interpret C strings as utf-8.
            # On PySide2 we patch QTextCodec.setCodecForCStrings() with a do-nothing stub,
            # so this will have no effect.
```